packages/compiler/test/checker/imports.test.ts

```diff
@@ -160,4 +166,150 @@ describe("compiler: imports", () => {
+              );
+            } else {
+              host.addTypeSpecFile(filename, JSON.stringify(fileConfig, null, 2));
+            }
+          }
+
+          await host.compile(config.entrypoint);
+          for (const [filename, expectedScope] of Object.entries(scopes)) {
+            const file = host.program.sourceFiles.get(resolveVirtualPath(filename));
+            ok(file, `Expected to have file "${filename}"`);
+            ok("scope" in file.file, `Expected file ${filename} to have a scope attached.`);
+            deepStrictEqual(file.file.scope, expectedScope);
+          }
+        },
+      };
+    }
+
+    const projectScope: ProjectScope = { type: "project" };
+    it("relative files are stays in project", async () => {
+      await givenStructure({
+        entrypoint: "my-project/main.tsp",
+        structure: {
+          "my-project/main.tsp": ["./other.tsp", "../common.tsp"],
+          "my-project/other.tsp": [],
+          "common.tsp": [],
+        },
+      }).expectScopes({
+        "my-project/main.tsp": projectScope,
+        "my-project/other.tsp": projectScope,
+        "common.tsp": projectScope,
+      });
+    });
+
+    it("importing a library resolve is as its library", async () => {
+      await givenStructure({
+        entrypoint: "my-project/main.tsp",
+        structure: {
+          "my-project/main.tsp": ["my-lib1", "my-lib2"],
+          "node_modules/my-lib1/package.json": {
+            name: "my-lib1",
+            tspMain: "./main.tsp",
+          },
+          "node_modules/my-lib1/main.tsp": [],
+          "node_modules/my-lib2/package.json": {
+            name: "my-lib2",
+            tspMain: "./main.tsp",
+          },
+          "node_modules/my-lib2/main.tsp": [],
+        },
+      }).expectScopes({
+        "my-project/main.tsp": projectScope,
+        "node_modules/my-lib1/main.tsp": { type: "library", name: "my-lib1" },
+        "node_modules/my-lib2/main.tsp": { type: "library", name: "my-lib2" },
+      });
+    });
+
+    it("library importing another library resolve each as their own scope", async () => {
+      await givenStructure({
+        entrypoint: "my-project/main.tsp",
+        structure: {
+          "my-project/main.tsp": ["my-lib1"],
+          "node_modules/my-lib1/package.json": {
+            name: "my-lib1",
+            tspMain: "./main.tsp",
+          },
+          "node_modules/my-lib1/main.tsp": ["my-lib2"],
+          "node_modules/my-lib2/package.json": {
+            name: "my-lib2",
+            tspMain: "./main.tsp",
+          },
+          "node_modules/my-lib2/main.tsp": [],
+        },
+      }).expectScopes({
+        "my-project/main.tsp": projectScope,
+        "node_modules/my-lib1/main.tsp": { type: "library", name: "my-lib1" },
+        "node_modules/my-lib2/main.tsp": { type: "library", name: "my-lib2" },
+      });
+    });
+
+    describe("if a file is imported from main project and library, FIRST loaded wins...", () => {
```

Comment:
This is the current behavior. It is not very nice to not have consistency here but at the same time this might be sign of a bad design from project structure. 

Alternative would be to always promote to the project scope if it is part of it in any way